modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/dbf/DbaseFileHeader.java

```diff
@@ -504,17 +524,6 @@ public int getHeaderLength() {
      * @throws IOException If errors occur while reading.
      */
     public void readHeader(ReadableByteChannel channel) throws IOException {
-        readHeader(channel, Charset.defaultCharset());
-    }
-
-    /**
-     * Read the header data from the DBF file.
-     *
-     * @param channel A readable byte channel. If you have an InputStream you need to use, you can
-     *     call java.nio.Channels.getChannel(InputStream in).
-     * @throws IOException If errors occur while reading.
-     */
-    public void readHeader(ReadableByteChannel channel, Charset charset) throws IOException {
```

Comment:
A public method in a public class cannot be removed just like this I'm afraid, we try to maintain backwards compatibility as much as possible. Two options:
- Add the method back
- Warn users upgrading of the API break in the [upgrade guide](https://github.com/geotools/geotools/blob/master/docs/user/welcome/upgrade.rst)

Both are acceptable to me.

Reply:
The `charset` is now a class field. I believe that keeping the method with same parameter will be confusing. Going to stick with second option.